cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/transformers/AbstractTransformerConversion.java

```diff
@@ -120,4 +121,80 @@ protected void addAliases(Identifiable<?> identifiable) {
             }
         }
     }
+
+    private static RegulatingControl getRegulatingControl(Context context, String regulatingControlId) {
+        return context.regulatingControlMapping().cachedRegulatingControls().get(regulatingControlId);
+    }
+
+    protected void addCgmesDataAsProperties(Identifiable<?> transformer, TapChanger rtc, TapChanger ptc, Context context) {
+        addRatioTapChangerProperties(transformer, rtc, context);
+        addPhaseTapChangerProperties(transformer, ptc, context);
+    }
+
+    private static void  addRatioTapChangerProperties(Identifiable<?> transformer, TapChanger rtc, Context context) {
+        if (rtc == null || rtc.getId() == null) {
+            return;
+        }
+
+        if (rtc.getRegulatingControlId() != null) {
+            String key = String.format("RatioTapChanger.%s.TapChangerControl", rtc.getId());
+            transformer.setProperty(key, rtc.getRegulatingControlId());
+        }
+
+        if (rtc.getHiddenCombinedTapChanger() != null) {
+            defineHiddenTapChangerProperties(transformer, rtc, rtc.getHiddenCombinedTapChanger(), "RatioTapChanger", context);
+        }
+    }
+
+    private static void  addPhaseTapChangerProperties(Identifiable<?> transformer, TapChanger ptc, Context context) {
+        if (ptc == null || ptc.getId() == null) {
+            return;
+        }
+
+        if (ptc.getRegulatingControlId() != null) {
+            String key = String.format("PhaseTapChanger.%s.TapChangerControl", ptc.getId());
+            transformer.setProperty(key, ptc.getRegulatingControlId());
+        }
+        if (ptc.getType() != null) {
+            String key = String.format("PhaseTapChanger.%s.type", ptc.getId());
+            transformer.setProperty(key, ptc.getType());
+        }
+
+        if (ptc.getHiddenCombinedTapChanger() != null) {
+            defineHiddenTapChangerProperties(transformer, ptc, ptc.getHiddenCombinedTapChanger(), "PhaseTapChanger", context);
+
+            String key = String.format("PhaseTapChanger.%s.type", ptc.getHiddenCombinedTapChanger().getId());
+            transformer.setProperty(key, ptc.getHiddenCombinedTapChanger().getType());
+        }
+    }
+
+    private static void defineHiddenTapChangerProperties(Identifiable<?> transformer, TapChanger tc, TapChanger hiddenTc, String propertyTag, Context context) {
```

Comment:
Why do we need all these properties ? We have most of these information in IIDM tap changer, no ?

Reply:
We have to use IIDM model in priority.

Reply:
@zamarrenolm How much does it happen?
Another thing: for a hidden tap changer, I would rather have an extension than a lot of properties...

Reply:
@zamarrenolm If it doesn't happen a lot **and** if there is no physical sense, we can afford to lose them (maybe with a warning?)